app/src/main/java/com/saucelabs/mydemoapp/android/view/fragments/QRFragment.java

```diff
@@ -68,56 +90,128 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
 
 	private void bindData() {
 		mDb = AppDatabase.getInstance(getActivity());
-		mScannerView = new ZXingScannerView(mAct);
-		binding.cameraFL.addView(mScannerView);
 		checkPermission();
 	}
 
 	private void checkPermission() {
 		if (ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
-			ActivityCompat.requestPermissions(requireActivity(), new String[]{Manifest.permission.CAMERA}, 1002);
+			requestPermissions( new String[]{Manifest.permission.CAMERA}, 1002);
 		} else {
 			TestFairy.addEvent("Location permission granted");
 		}
-		//ActivityCompat.requestPermissions(getActivity(), new String[]{Manifest.permission.CALL_PHONE}, MY_PERMISSIONS_REQUEST_READ_CONTACTS);
-		//MenuFragment.requestPermissions( new String[]{Manifest.permission.CALL_PHONE}, MY_PERMISSIONS_REQUEST_READ_CONTACTS)
 	}
 
 	@Override
 	public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
 		super.onRequestPermissionsResult(requestCode, permissions, grantResults);
 		if (requestCode == 1002) {
-			if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+
+			if (allPermissionsGranted()) {
 				TestFairy.addEvent("Camera permission granted");
+				cameraProvider();
 			} else {
 				TestFairy.addEvent("Camera permission denied");
 				ST.startMainActivity(mAct, ST.getBundle(FRAGMENT_PRODUCT_CATAlOG, 1));
-//                        singleton.showCommonDialog(null, "Permission Denied!", "Please go to settings and enable permissions.", getString(R.string.ok), true);
 			}
 		}
 	}
 
 	@Override
 	public void onPause() {
 		super.onPause();
-		mScannerView.stopCamera();           // Stop camera on pause
 	}
 
 	@Override
 	public void onResume() {
 		super.onResume();
-		mScannerView.setResultHandler(this); // Register ourselves as a handler for scan results.
-		mScannerView.startCamera();          // Start camera on resume
+		if (!allPermissionsGranted()) {
+			ActivityCompat.requestPermissions(requireActivity(), REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS);
```

Comment:
```suggestion
			requestPermissions(REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS);
```

Reply:
@diegoperini - requestPermissions from Fragment is deprecated :-( . Let me know if to change to something else

Reply:
Also in LocationFragment we are using "ActivityCompat": 
        ActivityCompat.requestPermissions(requireActivity(), new String[]{
                Manifest.permission.ACCESS_COARSE_LOCATION,
                Manifest.permission.ACCESS_FINE_LOCATION}, PERMISSION_ID);

Reply:
I tried this change, but I can't go to the fragment after this change

Reply:
Then you can keep it as is. Also keep the one in `LocationFragment`.

Reply:
That's what I did :-)